### DISCUSSION TITLE
Inconsistent headings in log in modal

### DISCUSSION THREAD

**Discussion by rhoadsstevens** [ManifoldScholar/manifold#3457] [Accessibility]
Priority level 2 (out of 3) from UW Accessible Technologies

WCAG mapping: 2.4.6 Headings and Labels

STEPS TO REPRODUCE:
1. While logged out, navigate to the profile icon on the home page and press enter to open the log in modal.
2. Examine the headings of the page with a screen reader or other tool.
3. Examine the heading structure for the "Forgot your password?" and "Need to sign up?" linked pages

OBSERVED BEHAVIOR:
There are no headings within the default page of  the log-in modal or the "forgot your password?" page. However, there is a heading 4 of "create an account" within the sign up page.

On the Edge instance, using NVDA to get the headers results in the headers from the main page (the content that should be behind the modal) to be announced.

EXPECTED BEHAVIOR:
As screen reader users rely on headings to communicate the structure and purpose of a page, a lack of headings forces the user to actively navigate the page to learn about the purpose and layout of the page. Thus, it would be beneficial to consistently have descriptive headings on all pages (ex. "Log in"). Additionally, as modals are their own layer of content, separate from the page behind it, modals should have their own heading structure starting at a heading 1.

**Comment by tsmyre** [comment]
**CIC Comment**
Action: investigate a bit further, but if implementation seems correct, let’s move on. There are inconsistencies between screen readers, just like browsers, and the best we can do is follow best practices around markup structure and ARIA usage.

**Washington Reply**
Agreed in this case. As it looks like best practices are being followed then that is all that can be done from the developer side.

**Comment by dananjohnson** [comment]
The inconsistency was resolved in v9. The modal dialog has an `<h1>` heading, and each view in the modal ("Log In", "Create Account", "Reset Password") begins with an `<h2>`.
